invoices/invoiceregistry_test.go

```diff
@@ -1006,27 +1006,34 @@ func TestInvoiceExpiryWithRegistry(t *testing.T) {
 	// Fwd time 1 day.
 	testClock.SetTime(testTime.Add(24 * time.Hour))
 
-	// Give some time to the watcher to cancel everything.
-	time.Sleep(500 * time.Millisecond)
-	if err := registry.Stop(); err != nil {
-		t.Fatalf("failed to stop invoice registry: %v", err)
-	}
-
 	// Create the expected cancellation set before the final check.
 	expectedCancellations = append(
 		expectedCancellations, invoicesThatWillCancel...,
 	)
 
-	// Retrospectively check that all invoices that were expected to be
-	// canceled are indeed canceled.
-	for i := range expectedCancellations {
-		invoice, err := registry.LookupInvoice(expectedCancellations[i])
-		if err != nil {
-			t.Fatalf("cannot find invoice: %v", err)
+	// canceled return a bool to indicate whether all the invoices are
+	// canceled.
+	canceled := func() bool {
+		canceled := false
+
+		for i := range expectedCancellations {
+			invoice, err := registry.LookupInvoice(
+				expectedCancellations[i],
+			)
+			require.NoError(t, err)
+
+			canceled = invpkg.ContractCanceled == invoice.State
```

Comment:
Do we need to use `canceled && invpkg.ContractCanceled == invoice.State`? Otherwise it's sufficient for the last one to be canceled for the whole predicate to return `true`.

Reply:
ah exactly! now fixed!